packages/amplify-e2e-core/src/utils/sdk-calls.ts

```diff
@@ -144,6 +145,20 @@ export const getUserPool = async (userpoolId, region) => {
   return res;
 };
 
+export const getIdentityPoolRoles = async (identityPoolId: string, region: string) => {
+  config.update({ region });
+
+  let res;
+
+  try {
+    res = await new CognitoIdentity().getIdentityPoolRoles({ IdentityPoolId: identityPoolId }).promise();
+  } catch (e) {
+    console.log(e);
+  }
```

Comment:
I noticed this is a pattern in this file. But is this a good practice?
What happens at the caller in this case? Does it handle undefined response gracefully?
What does undef response mean for the caller ? If it's handled gracefully does this mean CLI would produce different outcomes if transient error happen?

I think we should either have retry loop with exponential backoff OR such behavior should be configured when we spin up CongnitoIdentity client. In either case transient errors. if they persist, should propagate and stop user's command.

Reply:
> I noticed this is a pattern in this file. But is this a good practice?

I don't think this is a good pattern because it swallows the error and does not surface errors that arise from SDK, including on errors from the caller payload. I also don't want this pattern to leak into the production code where it's called from the CLI.

> What does undef response mean for the caller ?

Since these are test helpers, I imagine the desired/expected behavior would be that the test fails, but again, we aren't surfacing the error, so there isn't a good understanding of why the test failed which can cause more pain.

> If it's handled gracefully does this mean CLI would produce different outcomes if transient error happen?

I don't know if this is relevant because this is just the test calling straight to the SDK (or at least that's how I am using it) and not going through CLI.

> I think we should either have retry loop with exponential backoff OR such behavior should be configured when we spin up CongnitoIdentity client. In either case transient errors. if they persist, should propagate and stop user's command.

I think we can do this, but I think this is out of scope for this PR. I also am not sure if failed calls to SDK are a huge pain point in tests. I would think that the issues with incorrect payloads and/or failing to match SDK responses cause test failures, which wouldn't be solved by retries.

Reply:
woah, I didn't notice this is tests. nvm me.

Reply:
But fwiw. I'd just remove try catch and let tests fail fast.